### PR TITLE
[eclipse/xtext-core#1183]: Fix tests on windows machines.

### PR DESCRIPTION
**Pull request by ArneDeutsch**
Signed-off-by: Arne Deutsch <Arne.Deutsch@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.common.types.tests/tests/org/eclipse/xtext/common/types/access/impl/AbstractTypeProviderTest.java [hunk 1]
```diff
@@ -3530,7 +3530,8 @@ public void testFindTypeByName_AbstractMultimap() {
 		String typeName = "com.google.common.collect.AbstractMultimap";
 		JvmGenericType type = (JvmGenericType) getTypeProvider().findTypeByName(typeName);
 		assertNotNull(type);
-		diagnose(type, "java:/Objects/javax.annotation.Nullable#javax.annotation.Nullable");
+		diagnose(type, "java:/Objects/javax.annotation.Nullable#javax.annotation.Nullable",
```
Comment (cdietrich) [inline org.eclipse.xtext.common.types.tests/tests/org/eclipse/xtext/common/types/access/impl/AbstractTypeProviderTest.java]: @ArneDeutsch this seem to be unrelated and caused by different classpaths.
thus i assume this will fail with gradle

Comment (szarnekow) [reply org.eclipse.xtext.common.types.tests/tests/org/eclipse/xtext/common/types/access/impl/AbstractTypeProviderTest.java]: Maybe we should exclude errorprone in the pom?

Comment (cdietrich) [reply org.eclipse.xtext.common.types.tests/tests/org/eclipse/xtext/common/types/access/impl/AbstractTypeProviderTest.java]: am not sure where this is coming from on arnes machine

Comment (cdietrich) [reply org.eclipse.xtext.common.types.tests/tests/org/eclipse/xtext/common/types/access/impl/AbstractTypeProviderTest.java]: will test this on my windows machine in the evening

Comment (cdietrich) [reply org.eclipse.xtext.common.types.tests/tests/org/eclipse/xtext/common/types/access/impl/AbstractTypeProviderTest.java]: @szarnekow i dont understand the test. there is a guard
`if (diagnostic.getSeverity() != Diagnostic.OK) {` in   `org.eclipse.xtext.common.types.access.impl.AbstractTypeProviderTest.diagnose(EObject, String...)  `
so that if there is no diagnostic at all the test will work regardless if there are any expected diagnostics or not.
is this behaviour intended? if no why does it behave differently on windows (or does it on your branch only @ArneDeutsch )

Comment (szarnekow) [reply org.eclipse.xtext.common.types.tests/tests/org/eclipse/xtext/common/types/access/impl/AbstractTypeProviderTest.java]: In different classpath configurations, we do have different missing transitive types. `diagnose(EObject, String...)` is a poor attempt to whitelist missing types.

Comment (szarnekow) [reply org.eclipse.xtext.common.types.tests/tests/org/eclipse/xtext/common/types/access/impl/AbstractTypeProviderTest.java]: It would be great if we could get rid of this.

**Comment by ArneDeutsch** [comment]
Not ready.
